src/hooks.rs

```diff
@@ -66,5 +66,12 @@ pub fn post_step_instruction(
 	_exec_scopes: &mut ExecutionScopes,
 	_constants: &HashMap<String, BigInt>,
 ) -> Result<(), VirtualMachineError> {
+	if *_vm.get_current_step() > _exec_scopes.get::<u64>(MAX_STEP)? as usize {
+		// TODO: find a better way to express custom errors
+		return Err(VirtualMachineError::CustomHint(
+			"max_step reached: ".to_string(),
+		));
+	}
+
```

Comment:
I would instead put this logic in the pre_step_instuction hook.
if you put max step to 3. You want to do steps 0, 1, and 2 successfully. Then beak before starting step 3

Reply:
Therefore you should use `>=` operator

Reply:
Assuming the cairoVM start counting steps at 0

Reply:
Also, put your logic in a function, and call this function in the hook. It will make hooks more readable as we stack logic in them